net/redisclient.go

```diff
@@ -222,11 +228,34 @@ func NewRedisRingClient(ro *RedisOptions) *RedisRingClient {
 		r.ring = redis.NewRing(ringOptions)
 		r.log = ro.Log
 		r.metricsPrefix = ro.MetricsPrefix
+
+		ch := make(chan []string)
+		go ro.AddrUpdater(r.quit, ch)
+		go func(quit chan struct{}, rCH <-chan []string) {
+			for {
+				select {
+				case <-quit:
+					return
+				case addrs := <-rCH:
+					r.SetAddrs(context.Background(), createAddressMap(addrs))
+				}
+			}
+		}(r.quit, ch)
+		r.ch = ch
 	}
 
 	return r
 }
 
+func createAddressMap(addrs []string) map[string]string {
+	res := make(map[string]string)
+	for idx, addr := range addrs {
+		addr = strings.TrimLeft(addr, "TCP://")
```

Comment:
Should be `TrimPrefix` I guess

Reply:
Also I think this trim is only needed for dynamic resize so maybe it is better done at the caller

Reply:
TrimLeft -> TrimPrefix

I don't think we need to optimize it and it seems more well to cleanup the data we get, but I can also move the cleanup to skipper.go.